src/Traits/HasMetaFields.php

```diff
@@ -18,20 +19,59 @@
     /**
      * @var array
      */
-    private $customMetaClasses = [
-        \Corcel\Model\Comment::class,
-        \Corcel\Model\Term::class,
-        \Corcel\Model\User::class,
+    private $builtInClasses = [
+        \Corcel\Model\Comment::class => \Corcel\Model\Meta\CommentMeta::class,
+        \Corcel\Model\Post::class    => \Corcel\Model\Meta\PostMeta::class,
+        \Corcel\Model\Term::class    => \Corcel\Model\Meta\TermMeta::class,
+        \Corcel\Model\User::class    => \Corcel\Model\Meta\UserMeta::class,
     ];
 
     /**
      * @return \Illuminate\Database\Eloquent\Relations\HasMany
      */
     public function meta()
     {
-        return $this->hasMany(
-            $this->getClassName(), $this->getFieldName()
-        );
+        return $this->hasMany($this->getMetaClass(), $this->getMetaForeignKey());
+    }
+
+    /**
+     * @return string
+     *
+     * @throws \UnexpectedValueException
+     */
+    protected function getMetaClass()
+    {
+        foreach ($this->builtInClasses as $model => $meta) {
+            if ($this instanceof $model) {
+                return $meta;
+            }
+        }
+
+        throw new UnexpectedValueException(sprintf(
+            '%s must extends one of Corcel built-in models: Comment, Post, Term or User.',
+            static::class
+        ));
+    }
+
+    /**
+     * @return string
+     *
+     * @throws \UnexpectedValueException
+     */
+    protected function getMetaForeignKey()
+    {
+        foreach ($this->builtInClasses as $model => $meta) {
```

Comment:
Let's use a Laravel collection here, what do you think? Maybe something like:
```php
collect($this->builtInClasses)->first(...)
```

Reply:
Laravel's `first` method is returning `$value` from array, there is no way to modify it in callable: https://github.com/laravel/framework/blob/5.5/src/Illuminate/Support/Arr.php#L174-L178. Maybe I could use `map` and `flip` to swap keys with values, but I think that this will make code less readable.

Reply:
@Dartui no problem then. can you only fix the conflict and push again? then I merge this PR ;-) thanks